### PR TITLE
build(static_centerline_generator): prefix package and namespace with autoware_

### PR DESCRIPTION
**Pull request by esteve**
## Description

This PR adds the `autoware_` prefix to the package and puts headers in the `autoware` namespace.

See https://github.com/autowarefoundation/autoware/issues/4569

## Tests performed

Ran the tests suite.

## Effects on system behavior

Not applicable.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [ ] I've confirmed the [contribution guidelines].
- [ ] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Review by takayuki5168 — APPROVED** [review]

File: planning/static_centerline_generator/src/static_centerline_generator_node.cpp [hunk 1]
```diff
@@ -431,14 +437,14 @@ std::vector<lanelet::Id> StaticCenterlineGeneratorNode::plan_route(
...
-    const auto route = mission_planner->plan(check_points);
+    auto route = mission_planner->plan(check_points);
 
     return route;
   }();
   RCLCPP_INFO(get_logger(), "Planned route.");
 
   // get lanelets
-  const auto route_lane_ids = get_lane_ids_from_route(route);
+  auto route_lane_ids = get_lane_ids_from_route(route);
```
Comment (takayuki5168) [inline planning/static_centerline_generator/src/static_centerline_generator_node.cpp:447]: Why did you remove const?

File: planning/static_centerline_generator/include/static_centerline_generator/centerline_source/optimization_trajectory_based_centerline.hpp [hunk 2]
```diff
@@ -12,32 +12,43 @@
...
-public:
-  OptimizationTrajectoryBasedCenterline() = default;
-  explicit OptimizationTrajectoryBasedCenterline(rclcpp::Node & node);
-  std::vector<TrajectoryPoint> generate_centerline_with_optimization(
-    rclcpp::Node & node, const RouteHandler & route_handler,
-    const std::vector<lanelet::Id> & route_lane_ids);
+  using ::autoware_auto_planning_msgs::msg::Path;
+  using ::autoware_auto_planning_msgs::msg::PathWithLaneId;
+  using ::autoware_auto_planning_msgs::msg::TrajectoryPoint;
+  using ::route_handler::RouteHandler;
```
Comment (takayuki5168) [inline planning/static_centerline_generator/include/static_centerline_generator/centerline_source/optimization_trajectory_based_centerline.hpp:35]: type_alias.hpp file is used for the list of `using`.
Any reason to add this list of `using` here?

File: planning/static_centerline_generator/src/centerline_source/optimization_trajectory_based_centerline.cpp [hunk 3]
```diff
@@ -116,7 +116,7 @@ OptimizationTrajectoryBasedCenterline::generate_centerline_with_optimization(
 }
 
 std::vector<TrajectoryPoint> OptimizationTrajectoryBasedCenterline::optimize_trajectory(
```
Comment (takayuki5168) [inline planning/static_centerline_generator/src/centerline_source/optimization_trajectory_based_centerline.cpp:118]: Any reason to remove const here?

File: planning/static_centerline_generator/src/centerline_source/optimization_trajectory_based_centerline.cpp [hunk 4]
```diff
@@ -116,7 +116,7 @@ OptimizationTrajectoryBasedCenterline::generate_centerline_with_optimization(
...
-  const Path & raw_path) const
```
Comment (takayuki5168) [inline planning/static_centerline_generator/src/centerline_source/optimization_trajectory_based_centerline.cpp:119]: Any reason to remove const here?

**Review by takayuki5168 — CHANGES_REQUESTED** [review]

**Comment by takayuki5168** [comment]
@esteve 
Thank you for the change.

I have one comment.
The package name's change and the refactoring had better be separated. Especially, the package name's change has a lot of code changes and has the potential to have a conflict soon, so the change should be merged as soon as possible.

When you create a new PR for the package name's change, I highly recommend not adding the refactoring change.

File: planning/static_centerline_generator/src/static_centerline_generator_node.cpp (hunk 1 above)
Comment (esteve) [reply planning/static_centerline_generator/src/static_centerline_generator_node.cpp:447]: It was a clang-tidy warning

**Comment by esteve** [comment]
> The package name's change and the refactoring had better be separated. Especially, the package name's change has a lot of code changes and has the potential to have a conflict soon, so the change should be merged as soon as possible.

I used the opportunity to fix the clang-tidy warnings that were causing the CI to fail, instead of ignoring them.

File: planning/static_centerline_generator/src/centerline_source/optimization_trajectory_based_centerline.cpp (hunk 4 above)
Comment (esteve) [reply planning/static_centerline_generator/src/centerline_source/optimization_trajectory_based_centerline.cpp:119]: Same as https://github.com/autowarefoundation/autoware.universe/pull/6817#discussion_r1567585660 do you want to revert the change despite the `clang-tidy` warning?

File: planning/static_centerline_generator/src/static_centerline_generator_node.cpp (hunk 1 above)
Comment (takayuki5168) [reply planning/static_centerline_generator/src/static_centerline_generator_node.cpp:447]: Do you know why the clang-tidy recommended it? Removing const does not make sense to me.

File: planning/static_centerline_generator/src/centerline_source/optimization_trajectory_based_centerline.cpp (hunk 4 above)
Comment (takayuki5168) [reply planning/static_centerline_generator/src/centerline_source/optimization_trajectory_based_centerline.cpp:119]: If there is no reason except for just clang-tidy, yes.

File: planning/static_centerline_generator/include/static_centerline_generator/centerline_source/optimization_trajectory_based_centerline.hpp (hunk 2 above)
Comment (takayuki5168) [reply planning/static_centerline_generator/include/static_centerline_generator/centerline_source/optimization_trajectory_based_centerline.hpp:35]: Yes, but I want to make the `using` clauses unified by adding them only in `type_alias.hpp`.
If you highly recommend adding them outside `type_alias.hpp`, we'd better remove `type_alias.hpp` to make them unified.

**Comment by mitsudome-r** [comment]
@esteve As we have discussed in the Software WG call, I think it's better to first focus on package name fix first, and you can ignore the clang-tidy issues.

**Comment by esteve** [comment]
@takayuki5168 I've reverted any performance or any other improvements detected by clang-tidy, the CI will now fail because of the clang-tidy check, but now the PR only contains the prefix changes.

**Review by takayuki5168 — APPROVED** [review]
Much easier to review. Thank you!

**Comment by esteve** [comment]
> @esteve Can you fill in the following blanks in the PR's description?
> 
> > Tests performed

Done.

> > Effects on system behavior

These changes do no affect the system behavior.

**Comment by xmfcx** [comment]
I've rebased this to the latest main to test it with the planning simulator. If it works, I will merge it.

**Comment by xmfcx** [comment]
I've went through the conversations in [here](https://github.com/orgs/autowarefoundation/discussions/4097#discussioncomment-8380000) but I have a question:

Why keep these consistent with the updated package names? (package parent folder and the folder under include)

Screenshot from 2024-05-02 17-17-01

I've checked https://github.com/moveit/moveit and all their package names correspond uses `moveit_` prefix.
In [moveit_core](https://github.com/moveit/moveit/tree/master/moveit_core) there are folders without prefix but they are not packages anyways.

So to keep it consistent we should prefix these folders with `autoware_` as well.

**Comment by xmfcx** [comment]
I've compiled it and ran the planning simulator. Then realized this is an offline package.

Created this discussion: [Move static_centerline_generator to autoware_tools #4691](https://github.com/orgs/autowarefoundation/discussions/4691)

This is a separate discussion and can be done once we merge this.

**Comment by esteve** [comment]
Back to draft because I'm going to rename folders and move headers to a separate folder (see https://github.com/autowarefoundation/autoware/issues/4569#issuecomment-2097690750)

**Comment by xmfcx** [comment]
@esteve this only needed a package folder rename, why move the headers to a new folder?

(The include directory is not copied to the install directory on this package)

**Comment by esteve** [comment]
@xmfcx I missed the part where the headers in `include` are not copied during install, which in that case, should be moved to `src` if they're not exported, like we do with other packages (e.g. https://github.com/autowarefoundation/autoware.universe/tree/main/planning/behavior_velocity_planner/src)

**Comment by esteve** [comment]
@xmfcx in fact, correctly if I'm wrong, but it does seem that the headers from `include` are copied during install, since this package uses `ament_auto_package` https://github.com/autowarefoundation/autoware.universe/blob/main/planning/static_centerline_generator/CMakeLists.txt#L39C1-L44 (which copies headers if `include` exists https://github.com/ament/ament_cmake/blob/humble/ament_cmake_auto/cmake/ament_auto_package.cmake#L36-L38)

**Comment by esteve** [comment]
@xmfcx this is now ready for review

**Comment by xmfcx** [comment]
https://github.com/autowarefoundation/autoware.universe/actions/runs/9077838750/job/24943591808?pr=6817#step:7:21145

```
/__w/autoware.universe/autoware.universe/planning/autoware_static_centerline_generator/src/centerline_source/optimization_trajectory_based_centerline.cpp:15:10: fatal error: centerline_source/optimization_trajectory_based_centerline.hpp: No such file or directory
```

Something is wrong with the header locations, the compilation failed on my machine, also on CI.

**Comment by esteve** [comment]
@xmfcx let me check, I must have forgotten to commit something. Regarding the precommit, I assumed that any cpplint issue was optional, but I can fix it anyway

File: planning/autoware_static_centerline_generator/launch/static_centerline_generator.launch.xml [hunk 5]
```diff
@@ -55,7 +55,7 @@
   </node>
 
   <!-- optimize path -->
-  <node pkg="static_centerline_generator" exec="main" name="static_centerline_generator">
+  <node pkg="autoware_static_centerline_generator" exec="main" name="autoware_static_centerline_generator">
```
Comment (xmfcx) [inline planning/autoware_static_centerline_generator/launch/static_centerline_generator.launch.xml:58]:
```suggestion
  <node pkg="autoware_static_centerline_generator" exec="main" name="static_centerline_generator">
```
https://autowarefoundation.github.io/autoware-documentation/main/contributing/coding-guidelines/ros-nodes/class-design/#gnss_poser_nodecpp

The default node name shouldn't have `autoware_` prefix.

File: planning/static_centerline_generator/src/main.cpp [hunk 6]
```diff
@@ -1,39 +0,0 @@
-// Copyright 2022 Tier IV, Inc.
```
[the remark this answers is not in the document]
Comment (xmfcx) [reply planning/static_centerline_generator/src/main.cpp]: In the latest iteration, the main.cpp is back: https://github.com/autowarefoundation/autoware.universe/pull/6817/files#diff-f2ad628996669c755c9205f9a99aacae0c401072866142aecb3fa95fa9fbe857

File: planning/static_centerline_generator/include/static_centerline_generator/centerline_source/optimization_trajectory_based_centerline.hpp (hunk 2 above)
Comment (xmfcx) [reply planning/static_centerline_generator/include/static_centerline_generator/centerline_source/optimization_trajectory_based_centerline.hpp]: In the latest iteration of this PR, type_alias is used instead of locally defining these.

**Comment by esteve** [comment]
@xmfcx I don't know what happened, but the changes for `static_centerline_generator` got mixed in. I've reverted them.

File: planning/autoware_static_centerline_generator/src/static_centerline_generator_node.cpp [hunk 7]
```diff
@@ -168,7 +169,7 @@ std::vector<TrajectoryPoint> resample_trajectory_points(
 
 StaticCenterlineGeneratorNode::StaticCenterlineGeneratorNode(
   const rclcpp::NodeOptions & node_options)
-: Node("static_centerline_generator", node_options)
+: Node("autoware_static_centerline_generator", node_options)
```
Comment (xmfcx) [inline planning/autoware_static_centerline_generator/src/static_centerline_generator_node.cpp:172]: Should be `static_centerline_generator`.

File: planning/autoware_static_centerline_generator/src/static_centerline_generator_node.cpp [hunk 8]
```diff
@@ -216,19 +217,19 @@ StaticCenterlineGeneratorNode::StaticCenterlineGeneratorNode(
...
   srv_plan_route_ = create_service<PlanRoute>(
-    "/planning/static_centerline_generator/plan_route",
+    "/planning/autoware_static_centerline_generator/plan_route",
     std::bind(
       &StaticCenterlineGeneratorNode::on_plan_route, this, std::placeholders::_1,
       std::placeholders::_2),
     rmw_qos_profile_services_default, callback_group_);
   srv_plan_path_ = create_service<PlanPath>(
-    "/planning/static_centerline_generator/plan_path",
+    "/planning/autoware_static_centerline_generator/plan_path",
```
Comment (xmfcx) [inline planning/autoware_static_centerline_generator/src/static_centerline_generator_node.cpp:232]: Should be without the prefix.

File: planning/autoware_static_centerline_generator/rviz/static_centerline_generator.rviz [hunk 9]
```diff
@@ -179,7 +179,7 @@ Visualization Manager:
         Filter size: 10
         History Policy: Keep Last
         Reliability Policy: Reliable
-        Value: /static_centerline_generator/output_centerline
+        Value: /autoware_static_centerline_generator/output_centerline
```
Comment (xmfcx) [inline planning/autoware_static_centerline_generator/rviz/static_centerline_generator.rviz:182]: Should be without prefix since the node name don't have the prefix too.

File: planning/autoware_static_centerline_generator/rviz/static_centerline_generator.rviz [hunk 10]
```diff
@@ -222,7 +222,7 @@ Visualization Manager:
             Filter size: 10
             History Policy: Keep Last
             Reliability Policy: Reliable
-            Value: /static_centerline_generator/debug/raw_centerline
+            Value: /autoware_static_centerline_generator/debug/raw_centerline
```
Comment (xmfcx) [inline planning/autoware_static_centerline_generator/rviz/static_centerline_generator.rviz:225]: Should be without prefix since the node name don't have the prefix too.

File: planning/autoware_static_centerline_generator/rviz/static_centerline_generator.rviz [hunk 11]
```diff
@@ -268,7 +268,7 @@ Visualization Manager:
             Durability Policy: Transient Local
             History Policy: Keep Last
             Reliability Policy: Reliable
-            Value: /static_centerline_generator/debug/unsafe_footprints
+            Value: /autoware_static_centerline_generator/debug/unsafe_footprints
```
Comment (xmfcx) [inline planning/autoware_static_centerline_generator/rviz/static_centerline_generator.rviz:271]: Should be without prefix since the node name don't have the prefix too.

File: planning/autoware_static_centerline_generator/scripts/app.py [hunk 12]
```diff
@@ -85,7 +85,7 @@ def post_planned_route():
         print("map_id is not correct.")
 
     # create client
-    cli = create_client(PlanRoute, "/planning/static_centerline_generator/plan_route")
+    cli = create_client(PlanRoute, "/planning/autoware_static_centerline_generator/plan_route")
```
Comment (xmfcx) [inline planning/autoware_static_centerline_generator/scripts/app.py:88]: Should be without prefix since the node name don't have the prefix too.

File: planning/autoware_static_centerline_generator/scripts/app.py [hunk 13]
```diff
@@ -123,7 +123,7 @@ def post_planned_path():
         print("map_id is not correct.")
 
     # create client
-    cli = create_client(PlanPath, "/planning/static_centerline_generator/plan_path")
+    cli = create_client(PlanPath, "/planning/autoware_static_centerline_generator/plan_path")
```
Comment (xmfcx) [inline planning/autoware_static_centerline_generator/scripts/app.py:126]: Should be without prefix since the node name don't have the prefix too.

File: planning/autoware_static_centerline_generator/scripts/centerline_updater_helper.py [hunk 14]
```diff
@@ -155,7 +155,7 @@ def __init__(self):
         transient_local_qos = QoSProfile(depth=1, durability=QoSDurabilityPolicy.TRANSIENT_LOCAL)
         self.sub_whole_centerline = self.create_subscription(
             Trajectory,
-            "/static_centerline_generator/output_whole_centerline",
+            "/autoware_static_centerline_generator/output_whole_centerline",
```
Comment (xmfcx) [inline planning/autoware_static_centerline_generator/scripts/centerline_updater_helper.py:158]: Should be without prefix since the node name don't have the prefix too.

**Comment by esteve** [comment]
@xmfcx sorry, I just realized that I had enabled auto-merge and this PR got merged before your latest comments. I'll submit a followup PR with the fixes.

**Comment by esteve** [comment]
@xmfcx I've submitted https://github.com/autowarefoundation/autoware.universe/pull/7028

**Comment by xmfcx** [comment]
Thank you, It's my mistake I didn't realized those small things in my previous review too.
